src/Text/Markdown/SlamDown/Syntax/FormField.purs

```diff
@@ -11,11 +16,43 @@ import Data.Maybe as M
 import Test.StrongCheck as SC
 import Test.StrongCheck.Gen as SC
 
+-- | Pairing an underlying value and a display form.
+type StringReprR a =
+  { value :: a
+  , display :: String
+  }
+
+newtype StringRepr a = StringRepr (StringReprR a)
```

Comment:
`value` and `label` would be the HTML terms for these too, although I'm totally fine with `display` too.